test/compliance/generate_fbank_data.py

```diff
@@ -0,0 +1,174 @@
+            'frame_shift': '%.4f' % (float(random.randint(1, WAV_LEN - 1)) / 16000 * 1000),
+            'high_freq': str(high_freq),
+            'htk_compat': generate_rand_boolean(),
+            'low_freq': str(low_freq),
+            'num_mel_bins': str(random.randint(4, 8)),
+            'preemphasis_coefficient': '%.2f' % random.random(),
+            'raw_energy': generate_rand_boolean(),
+            'remove_dc_offset': generate_rand_boolean(),
+            'round_to_power_of_two': generate_rand_boolean(),
+            'snip_edges': generate_rand_boolean(),
+            'subtract_mean': generate_rand_boolean(),
+            'use_energy': generate_rand_boolean(),
+            'use_log_fbank': generate_rand_boolean(),
+            'use_power': generate_rand_boolean(),
+            'vtln_high': str(vtln_high),
+            'vtln_low': str(vtln_low),
+            'vtln_warp': '%.4f' % (vtln_warp_factor),
+            'window_type': generate_rand_window_type()
+        }
+
+        fn = 'fbank-' + ('-'.join(list(inputs.values())))
+        out_fn = OUTPUT_DIR + fn + '.ark'
+        arg = [
+            EXE_PATH,
+            '--blackman-coeff=' + inputs['blackman_coeff'],
+            '--dither=0.0',
+            '--energy-floor=' + inputs['energy_floor'],
+            '--frame-length=' + inputs['frame_length'],
+            '--frame-shift=' + inputs['frame_shift'],
+            '--high-freq=' + inputs['high_freq'],
+            '--htk-compat=' + inputs['htk_compat'],
+            '--low-freq=' + inputs['low_freq'],
+            '--num-mel-bins=' + inputs['num_mel_bins'],
+            '--preemphasis-coefficient=' + inputs['preemphasis_coefficient'],
+            '--raw-energy=' + inputs['raw_energy'],
+            '--remove-dc-offset=' + inputs['remove_dc_offset'],
+            '--round-to-power-of-two=' + inputs['round_to_power_of_two'],
+            '--snip-edges=' + inputs['snip_edges'],
+            '--subtract-mean=' + inputs['subtract_mean'],
+            '--use-energy=' + inputs['use_energy'],
+            '--use-log-fbank=' + inputs['use_log_fbank'],
+            '--use-power=' + inputs['use_power'],
+            '--vtln-high=' + inputs['vtln_high'],
+            '--vtln-low=' + inputs['vtln_low'],
+            '--vtln-warp=' + inputs['vtln_warp'],
+            '--window-type=' + inputs['window_type'],
+            SCP_PATH,
+            out_fn
+        ]
+
+        print(fn)
+        print(inputs)
+        print(' '.join(arg))
+
+        try:
+            if VERBOSE:
+                subprocess.call(arg)
+            else:
+                subprocess.call(arg, stderr=open(os.devnull, 'wb'), stdout=open(os.devnull, 'wb'))
+            # kaldi_output_dict = {k: v for k, v in torchaudio.kaldi_io.read_mat_ark(out_fn)}
+            # print(kaldi_output_dict)
+            print('success')
+        except Exception:
+            if os.path.exists(out_fn):
+                os.remove(out_fn)
+
+
+def parse(token):
+    if token == 'true':
```

Comment:
This parse function returns different type of results (bool, int, float). It may be a better way to parse the token.

Reply:
since the token is used in keyword arguments to `fbank`, they are different types. is there a better way of doing this?